File: test.spritebuilder/Source/TDEngine/Map/MapDAOmemoryImpl.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Map;


import Utils.Point2D;
import internalObject.SolderInternal.SolderType;
 
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

/**
 *
 * @author ver
 */
public class MapDAOmemoryImpl extends MapDAO {

    @Override
    public LevelDTO getMap(int ID_LEVEL) {
        
       List<Point2D > pathPoint= new ArrayList<Point2D  >();
       
        pathPoint.add(new Point2D  (0,0));
        pathPoint.add(new Point2D  (40,40));
        pathPoint.add(new Point2D  (40,50));
        pathPoint.add(new Point2D  (100,100));
      
      
        
         List<Point2D  > places= new ArrayList<Point2D  >();
          places.add(new Point2D (18,50));
        places.add(new Point2D (20,50));
        places.add(new Point2D (20,60));
        places.add(new Point2D (20,70)); 
       places.add(new Point2D (30,50));
       
       return new LevelDTO(pathPoint,places,getSolders(1));
    }

    public LevelDTO getMapWithMassivLoad( ) {
           List<Point2D > pathPoint= new ArrayList<Point2D  >();
       
        pathPoint.add(new Point2D  (0,0));
        pathPoint.add(new Point2D  (40,40));
        pathPoint.add(new Point2D  (40,50));
        pathPoint.add(new Point2D  (1000,1000));
        pathPoint.add(new Point2D  (2000,2000));
 

     
        
        
        
        List<Point2D  > places= new ArrayList<Point2D  >();
        places.add(new Point2D (18,50));
        places.add(new Point2D (20,50));
        places.add(new Point2D (20,60));
        places.add(new Point2D (20,70)); 
        places.add(new Point2D (30,50));
        places.add(new Point2D (90,90));
        places.add(new Point2D (900,1000));
        places.add(new Point2D (1000,1100));
        places.add(new Point2D (1100,1200));
       
       return new LevelDTO(pathPoint,places,getSolders(1000));
    }

    private List<UnitGroup> getSolders(int i) {
       List<UnitGroup> lst = new ArrayList<UnitGroup> ();
       
        for (int j = 0; j < i; j++) { 
             if(j>1)   
                 lst.add(getUnitGroupe2());
             else
                 lst.add(getUnitGroupe()); 
        }
       
       
       return lst; 
    }
    
    
    private UnitGroup getUnitGroupe2( ) {
        UnitGroup un = new UnitGroup();
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder1));
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder2));
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder3));
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder1));
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder2));
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder3));
        return un;
    }
    private UnitGroup getUnitGroupe( ) {
        UnitGroup un = new UnitGroup();
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder1));
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder1));
        un.deatachments.add(DetachmentFabric.getDetachment(Detachment.SizeGroupe.big, SolderType.Solder1));
        return un;
    }
    
    
}

File: TowerDefence.spritebuilder/Map/MapDAOmemoryImpl.swift
//
//  MapDAOmemoryImpl.swift
//  TowerDefence
//
//  Created by koveres on 06.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation


public class MapDAOmemoryImpl:MapDAO {
    
    
    public func getMap( ID_LEVEL:CInt) -> (LevelDTO) {
        
        var pathPoint:[CGPoint] = [CGPoint]();
        
        
        pathPoint.append(CGPoint(x:0,y:0));
        pathPoint.append(CGPoint(x:40,y:40));
        pathPoint.append(CGPoint(x:40,y:50));
        pathPoint.append(CGPoint(x:100,y:100));
        
        
        
        var places:[CGPoint] = [CGPoint]();
        places.append(CGPoint(x:18,y:50));
        places.append(CGPoint(x:20,y:50));
        places.append(CGPoint(x:20,y:60));
        places.append(CGPoint(x:20,y:70));
        places.append(CGPoint(x:30,y:50));
  
        return  LevelDTO(path1:pathPoint,placesToSetTower1:places,solders1:getSolders(1));
    }
    
    public func getMapWithMassivLoad( ) -> (LevelDTO) {
        var pathPoint:[CGPoint] = [CGPoint]();
        
        pathPoint.append(CGPoint(x:0,y:0));
        pathPoint.append(CGPoint(x:40,y:40));
        pathPoint.append(CGPoint(x:40,y:50));
        pathPoint.append(CGPoint(x:1000,y:1000));
        pathPoint.append(CGPoint(x:2000,y:2000));
        
        
        var places:[CGPoint] = [CGPoint]();
        places.append(CGPoint(x:18,y:50));
        places.append(CGPoint(x:20,y:50));
        places.append(CGPoint(x:20,y:60));
        places.append(CGPoint(x:20,y:70));
        places.append(CGPoint(x:30,y:50));
        places.append(CGPoint(x:90,y:90));
        places.append(CGPoint(x:900,y:1000));
        places.append(CGPoint(x:1000,y:1100));
        places.append(CGPoint(x:1100,y:1200));
        
        return  LevelDTO(path1:pathPoint,placesToSetTower1:places,solders1:getSolders(1000));
 
    }
    
    private func getSolders( i:Int) -> ([UnitGroup]) {
        var lst = [UnitGroup]();
        
        for var j = 0; j < i; ++j {
            if(j>1)
            {
                lst.append(getUnitGroupe2());
            }
            else
            {
                lst.append(getUnitGroupe());
            }
        }
        
       
        return lst;
    }
    
    
    private func getUnitGroupe2( ) -> (UnitGroup) {
      
        var deatachments:[Detachment] = [Detachment]();
        
        
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder1));
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder2));
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder3));
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder1));
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder2));
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder3));
    
        var un:UnitGroup =   UnitGroup(dtchs:deatachments);
        return un;
    }
    
    private func getUnitGroupe( ) -> (UnitGroup) {
       
        var deatachments:[Detachment] = [Detachment]();
        
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder1));
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder1));
        deatachments.append(DetachmentFabric.getDetachment(SizeGroupe.big, type_:SolderType.Solder1));
        
        
        var un:UnitGroup =   UnitGroup(dtchs:deatachments);
        return un;
    }
    
    
}
